### PR TITLE
RD-5205 Make headers consistent

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: https://cloudifysource.atlassian.net/browse/RD-5205

## Screenshots / Videos
**Before:**
Screen Shot 2022-07-11 at 10 13 38

**After:**
Screen Shot 2022-07-11 at 10 14 02

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [ ] I verified that all tests and checks have passed.
- [ ] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

latest:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3113/pipeline

https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3086/pipeline

 

## Documentation
TODO

### CODE REVIEW

**Review by qooban — COMMENTED** [review]
Did you look at all widgets? 🤔 According to comment https://cloudifysource.atlassian.net/browse/RD-5205?focusedCommentId=76508 we should try fixing all places. At least I found that Events/Logs and Cluster Status widgets are to be updated as well.

Please check CI as it is failing.

**Comment by leon-good-life** [comment]
I started system tests on Jenkins:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3086/pipeline

**Review by qooban — COMMENTED** [review]
Added some comments.

Seems like Deployments View widget and Events/Logs widget tests are failing in your test execution.

File: app/translations/en.json [hunk 1]
```diff
@@ -14,11 +14,11 @@
             "noServices": "No services available"
         },
         "servicesList": {
-            "modeName": "Node Name",
+            "modeName": "Node name",
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply app/translations/en.json]: fixed: b315f3a6c8523e2c5713a6b5b7e7b6a92ffc0f69

File: widgets/agents/src/AgentsTable.tsx [hunk 2]
```diff
@@ -157,15 +174,3 @@ export default class AgentsTable extends React.Component {
         );
     }
 }
-
-AgentsTable.propTypes = {
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/agents/src/AgentsTable.tsx]: I undid it: 1a301db63e85e2681fb9e78529ddd98fdbf12462

File: widgets/agents/src/props/AgentsPropType.ts [hunk 3]
```diff
@@ -1,3 +1,14 @@
+export type Agents = {
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/agents/src/props/AgentsPropType.ts]: I moved it: 4528bfa5e93c73ea534e89950a379c8cae32efeb

File: widgets/events/src/EventsTable.tsx [hunk 4]
```diff
@@ -149,7 +149,7 @@ export default class EventsTable extends React.Component {
                         }
                     />
                     <DataTable.Column
-                        label="Deployment Id"
+                        label="Deployment id"
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/events/src/EventsTable.tsx]: I renamed it: d8cd4cc84b4912bd35197afaea117da5a48500df

**Comment by leon-good-life** [comment]
I started system tests on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3095/pipeline

File: widgets/secrets/src/props/SecretPropType.ts [hunk 5]
```diff
@@ -1,3 +1,18 @@
+export type Secret = {
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/secrets/src/props/SecretPropType.ts]: There is no other place that defines `Secret` in `Cloudify-Stage`. I tried to find `is_hidden_value` (a field of `Secret`) in the repo. And I tried to find `type Secret` `interface Secret`, etc.

File: app/components/shared/cluster/ClusterServicesList.tsx [hunk 6]
```diff
@@ -31,7 +31,7 @@ const ClusterServicesList: FunctionComponent = () => {
     return (
         <DataTable noDataMessage={noServicesMessage} noDataAvailable={_.isEmpty(services)} selectable>
             <DataTable.Column label={i18n.t('cluster.servicesList.serviceType', 'Service Type')} width="25%" />
-            <DataTable.Column label={i18n.t('cluster.servicesList.modeName', 'Node Name')} width="25%" />
+            <DataTable.Column label={i18n.t('cluster.servicesList.nodeName', 'Node Name')} width="25%" />
```
Comment (qooban) [inline app/components/shared/cluster/ClusterServicesList.tsx:34]: Optional:

According to our recent agreement in the UI team about removing translation fallbacks, you can update `i18n.t` calls in this file.

Ref. [UI weekly agenda](https://docs.google.com/document/d/1hcxYa4NVoBp7jkKO9KnRrg9FfYugeEEasKpvJHTAHCQ)

File: widgets/events/src/EventsTable.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (qooban) [reply widgets/events/src/EventsTable.tsx]: Unresolving this conversation as I see that you did a change in one place, but it applies to all uses of id abbrev. 

See my original comment.

File: widgets/events/src/EventsTable.tsx [hunk 7]
```diff
@@ -159,12 +159,12 @@ export default class EventsTable extends React.Component {
                         }
                     />
                     <DataTable.Column
-                        label="Node Id"
+                        label="Node id"
                         name="node_name"
                         show={!this.isOneElementLike(data.nodeInstanceId) && fieldsToShow.includes('Node Id')}
```
Comment (qooban) [inline widgets/events/src/EventsTable.tsx:164]: Does it work properly after your change?

I suspect that the value provided for `fieldsToShow.includes` should be the same as the label or am I missing something? 

This applies to all uses of `fieldsToShow` concept in all modified files.

Also optionally nice to have would be to add translation for this file as you did for other files 🙂 It's optional from my perspective, so I'm leaving the decision to you.

File: widgets/agents/src/AgentsTable.tsx [hunk 8]
```diff
@@ -95,12 +116,15 @@ export default class AgentsTable extends React.Component {
...
                     />
-                    <DataTable.Column label="System" show={fieldsToShow.indexOf('System') >= 0} />
-                    <DataTable.Column label="Version" show={fieldsToShow.indexOf('Version') >= 0} />
-                    <DataTable.Column label="Install Method" show={fieldsToShow.indexOf('Install Method') >= 0} />
+                    <DataTable.Column label={t('columns.system')} show={fieldsToShow.indexOf('System') >= 0} />
+                    <DataTable.Column label={t('columns.version')} show={fieldsToShow.indexOf('Version') >= 0} />
+                    <DataTable.Column
+                        label={t('columns.installMethod')}
+                        show={fieldsToShow.indexOf('Install Method') >= 0}
+                    />
```
Comment (qooban) [inline widgets/agents/src/AgentsTable.tsx:127]: I'm wondering - can we have:

```suggestion
                    <DataTable.Column label={t('columns.system')} show={fieldsToShow.indexOf(t('columns.system')) >= 0} />
                    <DataTable.Column label={t('columns.version')} show={fieldsToShow.indexOf(t('columns.system')) >= 0} />
                    <DataTable.Column
                        label={t('columns.installMethod')}
                        show={fieldsToShow.indexOf(t('columns.system')) >= 0}
                    />
```
? WDYT? 🤔 

It applies to all places with `fieldsToShow` in modified files.

Comment (leon-good-life) [reply widgets/agents/src/AgentsTable.tsx]: [c89154c](https://github.com/cloudify-cosmo/cloudify-stage/pull/2050/commits/c89154c78b19c154ff6891dad81834aa1b4de21e)

File: widgets/events/src/EventsTable.tsx (hunk 7 above)
Comment (leon-good-life) [reply widgets/events/src/EventsTable.tsx]: I added translations to `EventsTable` including `fieldsToShow`: [f91cd9a](https://github.com/cloudify-cosmo/cloudify-stage/pull/2050/commits/f91cd9a077acc064f5e85768a286d76aa5b3819a)

File: app/components/shared/cluster/ClusterServicesList.tsx (hunk 6 above)
Comment (leon-good-life) [reply app/components/shared/cluster/ClusterServicesList.tsx]: I removed fallbacks from translations: 
646995e

**Comment by leon-good-life** [comment]
I started system tests: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3113/pipeline

**Review by qooban — COMMENTED** [review]
OK, I'll review once you fix the tests.

**Comment by leon-good-life** [comment]
I fixed `events_spec` and started it on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3115/pipeline

**Comment by leon-good-life** [comment]
@qooban What do you think if I will split this PR into multiple PRs, like a PR for EventsTable?

**Comment by leon-good-life** [comment]
I split this PR into 4 PRs:
RD-5205 part 1: make headers consistent in events widget #2061
RD-5205 part 2: make headers consistent in agents widget #2062
RD-5205 part 3: make headers consistent in Secrets widget #2063
RD-5205 part 4: make headers consistent in other tables #2064

**Pull request closed without merging**